pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -0,0 +1,350 @@
+	ApicastManagementAPI *string `json:"managementAPI,omitempty"`
+	// +optional
+	OpenSSLVerify *bool `json:"openSSLVerify,omitempty"`
+	// +optional
+	IncludeResponseCodes *bool `json:"responseCodes,omitempty"`
+	// +optional
+	RegistryURL *string `json:"registryURL,omitempty"`
+	// +optional
+	Image *string `json:"image,omitempty"`
+}
+
+type BackendSpec struct {
+	// +optional
+	Image *string `json:"image,omitempty"`
+
+	// +optional
+	RedisImage *string `json:"redisImage,omitempty"`
+}
+
+type SystemSpec struct {
+	// +optional
+	Image *string `json:"image,omitempty"`
+
+	// +optional
+	MemcachedImage *string `json:"memcachedImage,omitempty"`
+
+	// +optional
+	RedisImage *string `json:"redisImage,omitempty"`
+
+	// TODO should this field be optional? We have different approaches in Kubernetes.
+	// For example, in v1.Volume it is optional and there's an implied behaviour
+	// on which one is the default VolumeSource of the ones available. However,
+	// on v1.Probe the Handler field is mandatory and says that one of the
+	// available values and only one should be specified (it mandates to write
+	// something)
+
+	// +optional
+	FileStorageSpec *SystemFileStorageSpec `json:"fileStorage,omitempty"`
+
+	// TODO should union fields be optional?
+
+	// +optional
+	DatabaseSpec *SystemDatabaseSpec `json:"database,omitempty"`
+}
+
+type SystemFileStorageSpec struct {
+	// Union type. Only one of the fields can be set.
+	// +optional
+	PVC *SystemPVCSpec `json:"persistentVolumeClaim,omitempty"`
+	// +optional
+	S3 *SystemS3Spec `json:"amazonSimpleStorageService,omitempty"`
+}
+
+type SystemPVCSpec struct {
+	// +optional
+	StorageClassName *string `json:"storageClassName,omitempty"`
+}
+
+type SystemS3Spec struct {
+	AWSBucket      string                  `json:"awsBucket"`
+	AWSRegion      string                  `json:"awsRegion"`
+	AWSCredentials v1.LocalObjectReference `json:"awsCredentialsSecret"`
+}
+
+type SystemDatabaseSpec struct {
+	// Union type. Only one of the fields can be set
+	// +optional
+	MySQL *SystemMySQLSpec `json:"mysql,omitempty"`
+	// +optional
+	PostgreSQL *SystemPostgreSQLSpec `json:"postgresql,omitempty"`
+}
+
+type SystemMySQLSpec struct {
+	// +optional
+	Image *string `json:"image,omitempty"`
+}
+
+type SystemPostgreSQLSpec struct {
+	// +optional
+	Image *string `json:"image,omitempty"`
+}
+
+type ZyncSpec struct {
+	// +optional
+	Image *string `json:"image,omitempty"`
+	// +optional
+	PostgreSQLImage *string `json:"postgreSQLImage,omitempty"`
+}
+
+type HighAvailabilitySpec struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+func init() {
+	SchemeBuilder.Register(&APIManager{}, &APIManagerList{})
+}
+
+// SetDefaults sets the default values for the APIManager spec and returns true if the spec was changed
+func (apimanager *APIManager) SetDefaults() (bool, error) {
+	var err error
+	changed := false
+	changed = apimanager.setAPIManagerCommonSpecDefaults()
+	changed = apimanager.setApicastSpecDefaults()
```

Comment:
The two above changed values are being ignored.

Reply:
good catch :+1: I'll fix it in a PR